### PR TITLE
Fix panic when ip address without scheme is used for graphite backend

### PR DESCRIPTION
**Pull request by johnewing1**
---

# Description

Fixes panic when an IP address without a scheme was used as the graphite host.
E.g.
```
[GraphiteConf]
	Host = "127.0.0.1:80"
```

Results in 
```
2020/06/30 17:45:29 enabling syslog
2020/06/30 17:45:29 info: migrate.go:138: checking migrations
2020/06/30 17:45:29 info: search.go:208: Loading last datapoints from redis
2020/06/30 17:45:29 info: search.go:215: Done
2020/06/30 17:45:29 info: check.go:562: check alert example.graphite start with now set to 2020-06-30 16:45:29.095031
2020/06/30 17:45:29 info: check.go:562: check alert example.opentsdb.os.high.cpu start with now set to 2020-06-30 16:45:29.095031
2020/06/30 17:45:29 info: check.go:562: check alert example.opentsdb.cpu.lookup start with now set to 2020-06-30 16:45:29.095031
<nil>2020/06/30 17:45:29 error: expr.go:148: Error: runtime error: invalid memory address or nil pointer dereference. Origin: Schedule: Alert Name: example.graphite. Expression: avg(graphite("*.cpu.*.cpu.user", "5m", "", "host..cpu")) > 1, Stack: goroutine 92 [running]:
runtime/debug.Stack(0x2271b60, 0x2254880, 0x35f4ba0)
	/usr/local/Cellar/go/1.13.7/libexec/src/runtime/debug/stack.go:24 +0x9d
bosun.org/cmd/bosun/expr.errRecover(0xc000169e58, 0xc000290f20)
```	

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

- [x] Unit test added which reproduces panic edge case. 

# Checklist:

- [x ] This contribution follows the project's [code of conduct](https://github.com/bosun-monitor/bosun/blob/master/CODE_OF_CONDUCT.md)
- [x] This contribution follows the project's [contributing guidelines](https://github.com/bosun-monitor/bosun/blob/master/CONTRIBUTING.md)
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: graphite/graphite.go [hunk 1]
```diff
@@ -56,7 +56,9 @@ func (r *Request) Query(host string, header http.Header) (Response, error) {
 		Path:     "/render/",
 		RawQuery: v.Encode(),
 	}
-	if u, _ := url.Parse(host); u.Scheme != "" && u.Host != "" {
+
+	u, err := url.Parse(host)
+	if err== nil && u.Scheme != "" && u.Host != "" {
```
Comment (muffix) [inline graphite/graphite.go:61]: Worth adding the missing space here?
```suggestion
	if err== nil && u.Scheme != "" && u.Host != "" {
```

Maybe we should also add the style check to the GitHub action. Not sure why it's not complaining. It should. 🤔

File: graphite/graphite_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+//NewTestClient returns *http.Client with Transport replaced to avoid making real calls
+func NewTestClient(fn RoundTripFunc) *http.Client {
+	return &http.Client{
+		Transport: RoundTripFunc(fn),
+	}
+}
+
+func TestUrlParsePanic(t *testing.T) {
```
Comment (muffix) [inline graphite/graphite_test.go:27]: Nice, great to see more test coverage. 👍

File: graphite/graphite.go (hunk 1 above)
Comment (johnewing1) [reply graphite/graphite.go]: Fixed the style issues

**Review by muffix — APPROVED** [review]
